pkg/api/validate/quota.go

```diff
@@ -0,0 +1,93 @@
+package validate
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sort"
+
+	mgmtcompute "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-03-01/compute"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/compute"
+)
+
+// checkQuota checks usage quotas vs. resources required by cluster before cluster creation
+func (dv *openShiftClusterDynamicValidator) validateQuotas(ctx context.Context, oc *api.OpenShiftCluster) error {
+	//prepare usage client
+	spp := &oc.Properties.ServicePrincipalProfile
+	r, err := azure.ParseResourceID(oc.ID)
+	if err != nil {
+		return err
+	}
+	conf := auth.NewClientCredentialsConfig(spp.ClientID, string(spp.ClientSecret), spp.TenantID)
+	authorizer, err := conf.Authorizer()
+	if err != nil {
+		return err
+	}
+	usageClient := compute.NewUsageClient(r.SubscriptionID, authorizer)
```

Comment:
in Dynamic(), create a usage client called spUsage next to spPermissions.  Use the spAuthorizer that already exists there instead of creating a new one.  Pass in spUsage, just like what happens in validateVnetPermissions.